skl2onnx/common/_container.py

```diff
@@ -177,25 +214,32 @@ def add_output(self, variable):
             self._outputs.append(variable)
 
 
-class ModelComponentContainer(ModelContainer):
+class ModelComponentContainer(ModelContainer, _WhiteBlackContainer):
     """
     In the conversion phase, this class is used to collect all materials
     required to build an *ONNX* *GraphProto*, which is encapsulated in a
     *ONNX* *ModelProto*.
     """
 
     def __init__(self, target_opset, options=None, dtype=None,
-                 registered_models=None):
+                 registered_models=None,
+                 white_op=None, black_op=None):
         """
         :param target_opset: number, for example, 7 for *ONNX 1.2*, and
                              8 for *ONNX 1.3*.
         :param dtype: float type to be used for every float coefficient
         :param options: see :ref:`l-conv-options`
         :param registered_models: registered models
+        :param white_op: white list of ONNX nodes allowed
+            while converting a pipeline, if empty, all are allowed
+        :param black_op: black list of ONNX nodes allowed
```

Comment:
Can you help me understand white_op and black_op? Do you exclude black_op nodes from model conversion?

Reply:
Some runtime do not implement all operators, a converter may know the list of available operators. The conversion raises an exception if one converter uses an operator not allowed. Actually, only the converter for GaussianMixture changes its behaviour not to use ReduceLogSumExp.